OpenMage/magento-lts
# [BC ]Unified CSRF configuration

Following https://github.com/OpenMage/magento-lts/pull/3146 I'd like to unify all the CSRF configuration in a single one, the actual situation is a bit confusing with three different flags for the same thing (general one, newsletter, checkout).

### Related Pull Requests
https://github.com/OpenMage/magento-lts/pull/3146

### Fixed Issues (if relevant)
Fixes the problem that some user may enabled CSRF only for some of the modules and not for others.

### Testing
- enable `system/csrf/use_form_key`
- check that both the newsletter forms and checkout have the form_key added to their `<form` element (actually the checkout does everything with ajax request so you'll have to check that the form_key is added to those requests in the developer console) as shown in this screenshot:
<img width="1455" alt="Screenshot 2023-04-05 alle 15 34 42" src="https://user-images.githubusercontent.com/909743/230114089-f9eee592-e61e-4bdd-b15a-5827004498bf.png">
